tutorials/smart-contracts/demo-aplications/deploying-uniswap-v2.md

```diff
@@ -0,0 +1,194 @@
+
+The project scaffolding is as follows:
+
+```bash
+uniswap-V2-polkadot
+├── bin/
+├── contracts/
+│   ├── interfaces/
+│   │   ├── IERC20.sol
+│   │   ├── IUniswapV2Callee.sol
+│   │   ├── IUniswapV2ERC20.sol
+│   │   ├── IUniswapV2Factory.sol
+│   │   └── IUniswapV2Pair.sol
+│   ├── libraries/
+│   │   ├── Math.sol
+│   │   ├── SafeMath.sol
+│   │   └── UQ112x112.sol
+│   ├── test/
+│   │   └── ERC20.sol
+│   ├── UniswapV2ERC20.sol
+│   ├── UniswapV2Factory.sol
+│   └── UniswapV2Pair.sol
+├── ignition/
+├── scripts/
+│   └── deploy.js
+├── node_modules/
+├── test/
+│   ├── shared/
+│   │   ├── fixtures.js
+│   │   └── utilities.js
+│   ├── UniswapV2ERC20.js
+│   ├── UniswapV2Factory.js
+│   └── UniswapV2Pair.js
+├── .env.example
+├── .gitignore
+├── hardhat.config.js
+├── package.json
+└── README.md
+```
+
+## Testing the Contracts
+
+You can run the provided test suite to ensure the contracts are working as expected. The tests cover various scenarios, including creating pairs, adding liquidity, and executing swaps.
+
+To test it locally, you can run the following commands:
+
+1. Spawn a local node for testing:
+
+    ```bash
+    npx hardhat node
+    ```
+
+    This command will spawn a local substrate node along with the eth-rpc adapter. The node will be available at `ws://127.0.0.1:8000` and the eth-rpc adapter at `http://localhost:8545`.
+
+2. In a new terminal, run the tests:
+
+    ```bash
+    npx hardhat test --network localNode
+    ```
+
+To test it on the Westend Hub, you can run the following command:
+
+```bash
+npx hardhat test --network westendHub
+```
+
+The result should look like this:
+
+--8<-- "code/tutorials/smart-contracts/demo-aplications/deploying-uniswap-V2/testing-output.html"
+
+## Deploying the Contracts
+
+After successfully testing the contracts, you can deploy them to the local node or the Polkadot Hub. The deployment script is located in the `scripts` directory and is named `deploy.js`. This script deploys the `Factory` and `Pair` contracts to the network.
+
+To deploy the contracts, run the following command:
+
+```bash
+npx hardhat run scripts/deploy.js --network localNode
+```
+
+or
+
+```bash
+npx hardhat run scripts/deploy.js --network westendHub
+```
+
+The deployment script will output the addresses of the deployed contracts. Save these addresses, as you will need them to interact with the contracts. For example, the output should look like this:
+
+--8<-- "code/tutorials/smart-contracts/demo-aplications/deploying-uniswap-V2/deployment-output.html"
+
+## Conclusion
+
+This tutorial taught you how to deploy Uniswap V2 contracts to Polkadot Hub, create liquidity pools, and perform token swaps. This implementation brings the powerful AMM model to the Polkadot ecosystem, enabling decentralized trading of any ERC-20 token pair.
```

Comment:
This is inaccurate as the guide does not explain how to create liquidity pools and perform token swaps. If that's intended, add a new section that describes how to interact with the protocol